pkg/endpointmanager/policymap_pressure.go

```diff
@@ -0,0 +1,65 @@
+package endpointmanager
+
+import (
+	"time"
+
+	"github.com/cilium/cilium/pkg/endpoint"
+	"github.com/cilium/cilium/pkg/maps/policymap"
+	"github.com/cilium/cilium/pkg/metrics"
+	"github.com/cilium/cilium/pkg/stream"
+	"github.com/cilium/cilium/pkg/trigger"
+)
+
+func (mgr *endpointManager) initPolicyMapPressureMetric() {
+	if !metrics.BPFMapPressure {
+		return
+	}
+	mgr.policyMapPressureGauge = metrics.NewBPFMapPressureGauge(policymap.MapName, 0.0)
+
+	go mgr.readPolicyMapPressureEvents()
+
+	var err error
+	src, emit, _ := stream.Multicast[endpoint.MapPressureEvent]() // ignoring complete() here as there's no single emitter that can complete it
+	mgr.policyMapPressureEventCh = stream.ToChannel(mgr.ctx, src)
```

Comment:
I would just do `src.Observe(mgr.ctx, mgr.onPolicyMapEvent, func(error) {})` here instead of going via a channel. Then one doesn't need to worry about draining the channel etc. concerns.

Reply:
Removed channel and usage of `pkg/stream`